pkg/net/libp2p/remoteconnection.go

```diff
@@ -224,9 +224,38 @@ func (conn *remoteConnection) process() {
 	defer close(conn.done)
 	defer conn.closeStream()
 
+	// Data to be sent to the connection.
+	// If nil, a new message from conn.msgBuffer will be read, encoded, and stored here.
+	var msgData []byte
+
 	for {
 		// The processing loop runs indefinitely (until interrupted by explicitly returning).
-		// One iteration corresponds to sending one message.
+		// One iteration corresponds to one attempt of sending a message.
+
+		// Check if connection is being closed.
+		// This is necessary for not getting stuck trying to send an unsent message
+		// (failing all the time and retrying forever).
+		select {
+		case <-conn.stop:
+			return
+		default:
+		}
+
+		// Get the next message and encode it if there is no pending unsent message.
+		if msgData == nil {
+			select {
+			case <-conn.stop:
+				return
+			case msg := <-conn.msgBuffer:
```

Comment:
Tests do not work because execution is blocked in this select and waits to receive a stop signal or msg.
The API contains `Connect`; because of that, as a user, I suggest that if I call `Connect` the peers are connected.
But you probably implicitly changed API by changing the implementation and now to connect a user must send a message.

Reply:
That's right, thanks!!
I've just fixed that. Now the underlying connections are established right after `Connect` is called (even though this usually doesn't finish before `Connect` returns).